src/install/lock.ts

```diff
@@ -288,7 +290,16 @@ export async function extractLockConstraintsAndMap (map: IImportMap, preloadUrls
           continue;
         }
       }
+
+      // Otherwise if the plain specifier is a reference to the base package
+      // containing the import map, we can safely ignore it, as the tracemap
+      // will always resolve such imports correctly at the top-level:
+      const pcfg = await resolver.getPackageConfig(pkgUrl);
+      if (pcfg && pcfg.name === parsedKey.pkgName) {
```

Comment:
We also need to handle subpath and environment deconstruction here like all other resolutions.